.circleci/config.yml

```diff
@@ -0,0 +1,61 @@
+jobs:
+  build38:
+    executor: 
+      name: python/default
+      tag: "3.8"
+    steps:
+      - checkout
+      - python/load-cache
+      - python/install-deps
+      - python/save-cache
+      - run:
+          command: |
+            python -m pytest --cov nway --cov-report xml
+            bash <(curl -s https://codecov.io/bash) -t ${CODECOV_TOKEN}
+          name: Test
+
+  lint:
+    executor: python/default
+    steps:
+      - checkout
+      - run:
+          command: |
+            pip install flake8
+            # `|| true` to force exit code 0 even if no files found
+            CHANGED_PYFILES=$(git diff --name-only --diff-filter AM origin/master | grep .py || true)
+            echo "List of changed files:"
+            echo ${CHANGED_PYFILES}
+            echo ${CHANGED_PYFILES} | xargs -r flake8 --count
+          name: Lint
+
+  docker:
+    machine: true
+    steps:
+      - checkout
+      - run: echo "$DOCKER_PASS" | docker login --username $DOCKER_USER --password-stdin
+      - run: 
+          name: Build and Push docker image
+          command: |
+            if [ "$CIRCLE_BRANCH" = "master" ]
+            then
+                MYTAG=master
+            else
+                MYTAG=develop
+            fi
+            image=alleninstitutepika/ophys_nway_matching:${MYTAG}
+            docker build --build-arg MYBRANCH=${CIRCLE_BRANCH} -t ${image} .
```

Comment:
I'm curious, does CircleCI cache this step? I could imagine building the image taking up a lot of time?

Reply:
addressed in earlier PR